torch/package/package_importer.py

```diff
@@ -624,6 +624,14 @@ def open_resource(self, resource):
     def resource_path(self, resource):
         # The contract for resource_path is that it either returns a concrete
         # file system path or raises FileNotFoundError.
+        if isinstance(
+            self.importer.zip_reader, DirectoryReader
+        ) and self.importer.zip_reader.has_record(
+            os.path.join(self.fullname, resource)
+        ):
+            return os.path.join(
+                self.importer.zip_reader.directory, self.fullname, resource
+            )
```

Comment:
I put this logic inside of this method because I didn't want to add it to the DirectoryReader itself. My thinking is that the DirectoryReader API should stay consistent with the PyTorchFileReader API, which doesn't have this functionality

Reply:
yep!